### PR TITLE
Implement TLS

### PR DESCRIPTION
**Pull request by stevesloka**
This adds TLS to v2.4.1 using (https://github.com/floragunncom/search-guard-ssl). There is one breaking change, I'm working to have this work with the helm chart here (https://github.com/kubernetes/charts/tree/master/incubator/elasticsearch) which switches the discovery service env variable. 

Happy to discuss! I'm going to have a PR to the helm charts repo to allow this to work with it.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,11 +6,18 @@ MAINTAINER pjpires@gmail.com
 ADD do_not_use.yml /elasticsearch/config/elasticsearch.yml
 
 # Install Elasticsearch plug-ins
-RUN /elasticsearch/bin/plugin install io.fabric8/elasticsearch-cloud-kubernetes/2.4.1 --verbose
+RUN /elasticsearch/bin/plugin install io.fabric8/elasticsearch-cloud-kubernetes/2.4.1
+RUN /elasticsearch/bin/plugin install lmenezes/elasticsearch-kopf
```
Comment (pires) [inline Dockerfile]: Why do we need this?

File: Dockerfile [hunk 2]
```diff
@@ -6,11 +6,18 @@ MAINTAINER pjpires@gmail.com
...
+RUN /elasticsearch/bin/plugin install com.floragunn/search-guard-ssl/2.4.1.16
 
 # Override elasticsearch.yml config, otherwise plug-in install will fail
 ADD elasticsearch.yml /elasticsearch/config/elasticsearch.yml
 
+# Copy in SSL Certs
+# NOTE: For a Kubernetes based deployment, make these secrets which are provided dynamically
+ADD scripts/node-keystore.jks /elasticsearch/config/node-keystore.jks
+ADD scripts/truststore.jks /elasticsearch/config/truststore.jks
+
```
Comment (pires) [inline Dockerfile]: Here we should do two things:
* ask for keystore and truststore
* pass them as volumes to container (will work both for Docker and for Kubernetes w/ secrets)

File: Dockerfile [hunk 3]
```diff
@@ -6,11 +6,18 @@ MAINTAINER pjpires@gmail.com
 ADD do_not_use.yml /elasticsearch/config/elasticsearch.yml
 
 # Install Elasticsearch plug-ins
-RUN /elasticsearch/bin/plugin install io.fabric8/elasticsearch-cloud-kubernetes/2.4.1 --verbose
+RUN /elasticsearch/bin/plugin install io.fabric8/elasticsearch-cloud-kubernetes/2.4.1
```
Comment (pires) [inline Dockerfile]: Here you install plug-in version 2.4.1 but below everything is 5.x related.

File: Dockerfile (hunk 2 above)
Comment (stevesloka) [inline Dockerfile]: That's a better approach.

**Pull request closed without merging**